include/powsybl/iidm/PhaseTapChangerHolder.hpp

```diff
@@ -22,10 +22,16 @@ class PhaseTapChangerHolder : public virtual TapChangerHolder {
 public:
     ~PhaseTapChangerHolder() noexcept override = default;
 
+    stdcxx::CReference<PhaseTapChanger> getOptionalPhaseTapChanger() const;
+
+    stdcxx::Reference<PhaseTapChanger> getOptionalPhaseTapChanger();
+
     virtual stdcxx::CReference<PhaseTapChanger> getPhaseTapChanger() const = 0;
 
     virtual stdcxx::Reference<PhaseTapChanger> getPhaseTapChanger() = 0;
 
+    bool hasPhaseTapChanger() const override;
```

Comment:
From a user point of view, it's really strange to have both `getOptionalPhaseTapChanger` and `getPhaseTapChanger` that return the same value. In java, we introduce the optional version for functional programming purpose.
2 options:
- only introduce the `hasPhaseTapChanger` method
- introduce optional methods, but change the return type of `getPhaseTapChanger` to return a reference (and throw an exception if not set even if it's different from the java version).

@gautierbureau What do you think?

Reply:
Personally I prefer first solution

Reply:
I would choose the 2nd one :D

Reply:
Updated in b064c3adbb2481d1634bff763453cb96fb448451 to implement the second one.